Examples/restcomm-olympus-nopod/restcomm-olympus/SigninTableViewController.m

```diff
@@ -118,36 +115,45 @@ - (RCDevice *)registerRCDevice{
     return [appDelegate registerRCDevice];
 }
 
-- (void)registerForPushWithAccount:(NSString *)account withRCDevice:(RCDevice *)rcDevice password:(NSString *)password andEmail:(NSString *)email{
-    //get token from user defaults
-    NSUserDefaults* appDefaults = [NSUserDefaults standardUserDefaults];
-    NSString *deviceToken = [appDefaults objectForKey:@"deviceToken"];
-    if (deviceToken && rcDevice){
-        //get certificate strings
-        NSString *pushCertificatesPathPublic = [[NSBundle mainBundle] pathForResource:@"certificate_key_push" ofType:@"pem"];
-        NSString *pushCertificatesPathPrivate = [[NSBundle mainBundle] pathForResource:@"rsa_private_key_push" ofType:@"pem"];
+- (void)registerForPush{
+    //before setting the push token, we need to check is push enabled (server can handle it)
+    if ([Utils isServerEnabledForPushNotifications]){
+        NSLog(@"Start registering for push on server");
+        NSString *deviceToken = [Utils pushToken];
+        NSString *pushAccount = [Utils pushAccount];
         
-      
-
-        NSMutableDictionary *dic = [[NSMutableDictionary alloc] initWithObjectsAndKeys:
-                                    @"Olympus", @"friendly-name",
-                                    account, @"username",
-                                    password, @"password",
-                                    email, @"rescomm-account-email",
-                                    deviceToken, @"token",
-                                    pushCertificatesPathPublic, @"push-certificate-public-path",
-                                    pushCertificatesPathPrivate, @"push-certificate-private-path",
-                                    [NSNumber numberWithBool:YES], @"is-sandbox", nil];
-        
-        [rcDevice registerPushToken:dic delegate:self];
-        
-    } else {
-        NSLog(@"Device Voip push token not found, or RCDevice not initialized");
+        if (deviceToken && pushAccount){
+            AppDelegate *appDelegate = ((AppDelegate *)[UIApplication sharedApplication].delegate);
+            RCDevice *rcDevice = [appDelegate registerRCDevice];
+            if (rcDevice){
+                //get certificate strings
+                NSString *pushCertificatesPathPublic = [[NSBundle mainBundle] pathForResource:@"certificate_key_push" ofType:@"pem"];
+                NSString *pushCertificatesPathPrivate = [[NSBundle mainBundle] pathForResource:@"rsa_private_key_push" ofType:@"pem"];
+                
+                
+                NSMutableDictionary *dic = [[NSMutableDictionary alloc] initWithObjectsAndKeys:
+                                            kFriendlyName, @"friendly-name",
+                                            [Utils pushAccount], @"rescomm-account-email",
+                                            [Utils pushPassword], @"password",
+                                            [Utils pushDomain], @"push-domain",
```

Comment:
Didn't we talk about introducing constant strings for all RCDevice related keys and use those instead in RCDevice APIs like RCDevice.registerPushToken() and RCDevice.initWithParams()?

Reply:
I dont follow. You mean, we should move all constants keys into sdk?

Reply:
Resolved per call